code/datums/datacore.dm

```diff
@@ -251,7 +256,166 @@ var/record_id_num = 1001
 		L.fields["b_dna"]		= H.dna.unique_enzymes
 		L.fields["enzymes"]		= H.dna.struc_enzymes
 		L.fields["identity"]	= H.dna.uni_identity
-		L.fields["image"]		= getFlatIcon(H)	//This is god-awful
+		L.fields["image"]		= image
 		locked += L
 	return
 
+/obj/effect/datacore/proc/get_id_photo(var/mob/living/carbon/human/H)
+	var/icon/photo = null
+	var/g = (H.gender == FEMALE) ? "f" : "m"
+	if(!config.mutant_races || H.dna.species.use_skintones)

```

Comment:
Are all of these species if() checks necessary? Or can there be one initial check in H.dna.species and branch from that?